### DISCUSSION TITLE
io.appium.settings's set_localtion can not offer mock location serivces to APP

### DISCUSSION THREAD

**Discussion by gbcbooks** [appium/appium#17764] [Q&A]
i try the following code
'adb -s 02157df2a809c10c shell am broadcast -a io.appium.settings.location -n io.appium.settings/.receivers.LocationInfoReceiver --ez forceUpdate false

adb -s 02157df2a809c10c shell appops set io.appium.settings android\:mock_location allow

adb -s 02157df2a809c10c shell  am startservice --user 0 -n io.appium.settings/.LocationService --es longitude 121.49054721208309 --es latitude 31.24065 --es altitude 2'

the location is The Bund of SHANGHAI, then i open BAIDU MAP(google map too), it is located to The Bund of SHANGHAI, the is match my expectation.
the next i open wechat , and try to send a location to a friend of mine, wechat did not locate to SHANGHAI , instead ,  it still using the real location.

also , i try 
'driver.set_location(31.24065, 121.49054721208309, 2) '
it does't work too. 
am i miss anything ?

**Comment by KazuCocoa** [comment]
Perhaps you could try https://github.com/appium/appium-uiautomator2-driver/#mobile-refreshgpscache to refresh the cached location data. The command had an issue before, but latest UIA2 driver version should work.

If the data applied to the device's google service depends on the OS. I observed such location update needed a few seconds by the OS. (this is Android OS's behavior, afaik)

**Reply by gbcbooks** [reply]
hello , i try appium2 server,  but it still can't set the location

**Reply by KazuCocoa** [reply]
Could you share the server log as well? We cannot say anything without logs. Probably the logcat also may help. Potentially it does not work well in some devices by manufacture customization etc.

**Reply by gbcbooks** [reply]
is it possible the Wechat skip google location service and use NMVE data from GPS hardware

**Reply by KazuCocoa** [reply]
`mockLocationApp` capability may help to use another mock location app you want. The default one is io.appium.settings, it depends on google's one.

**Reply by gbcbooks** [reply]
> Could you share the server log as well? We cannot say anything without logs. Probably the logcat also may help. Potentially it does not work well in some devices by manufacture customization etc.

i upload a server log as well , please check here https://gist.github.com/gbcbooks/c7790ef63c48e117ca730a433c8dabe6

**Reply by gbcbooks** [reply]
wel you can see , there is only info of setting location successful , i don't know where i can upload a screenshot of my phone to let you see that , it work on BAIDU map , but doesn't work on wechat

**Reply by KazuCocoa** [reply]
> it work on BAIDU map , but doesn't work on wechat

Not sure what the wechat refers to, but if it did not refer to the Google's location API (e.g. their own logic), the mock may not work

**Reply by gbcbooks** [reply]
> > it work on BAIDU map , but doesn't work on wechat
> 
> Not sure what the wechat refers to, but if it did not refer to the Google's location API (e.g. their own logic), the mock may not work

i think so, but  there is some app which provide somethink like sandbox function to lanuch wechat, after that , it can provide mock location to  wechat , so i think wechat much have a geolocatoin source provider, i am think  if i can find out what else location source can it be .
